crates/derive/src/traits/data_sources.rs

```diff
@@ -23,6 +23,31 @@ pub trait BlobProvider {
     ) -> Result<Vec<Box<Blob>>, Self::Error>;
 }
 
+/// The AltDAProvider trait specifies the functionality of a data source that can provide altda blobs.
+#[async_trait]
+pub trait AltDAProvider {
+    /// The error type for the [EigenDAProvider].
+    type Error: Display + ToString + Into<PipelineErrorKind>;
+
+    async fn get_blob(&self, commitment: Bytes) -> Result<Bytes, Self::Error>;
+}
+
+/// The EigenDAProvider trait specifies the functionality of a data source that can provide eigenda blobs.
+#[async_trait]
+pub trait EigenDAProvider {
```

Comment:
why do we define a trait for EigenDA, would this be one impl for AltDAProvider?

Reply:
Prob only so I can create a test implementation for testing.